Gems/PhysX/Code/Include/PhysX/UserDataTypes.inl

```diff
@@ -23,38 +22,26 @@ namespace PhysX
         actor->userData = this;
     }
 
-    inline BaseActorData::BaseActorData(BaseActorData&& other) :
-        m_sanity(s_sanityValue), m_actorType(other.m_actorType), m_actor(AZStd::move(other.m_actor))
+    inline ActorData::ActorData(ActorData&& other)
+        : m_sanity(s_sanityValue)
+        , m_actor(AZStd::move(other.m_actor))
     {
         m_actor->userData = this;
     }
 
-    inline BaseActorData& BaseActorData::operator=(BaseActorData&& other)
+    inline ActorData& ActorData::operator=(ActorData&& other)
```

Comment:
Now that move operator is done by ActorData it's missing to move the `m_payload` member, which before it was done because it was using the default implementation. Same for the move constructor.

Reply:
good catch! added moving m_payload too. thanks!